src/allmydata/test/web/common.py

```diff
@@ -1,6 +1,18 @@
 
+import re
+
 unknown_rwcap = u"lafs://from_the_future_rw_\u263A".encode('utf-8')
 unknown_rocap = u"ro.lafs://readonly_from_the_future_ro_\u263A".encode('utf-8')
 unknown_immcap = u"imm.lafs://immutable_from_the_future_imm_\u263A".encode('utf-8')
 
 FAVICON_MARKUP = '<link href="/icon.png" rel="shortcut icon" />'
+
+def assert_soup_has_favicon(testcase, soup):
+    links = soup.find_all('link', rel='shortcut icon')
```

Comment:
docstrings for _all_ new APIs please :)  I know the codebase currently sets a poor example in that regard. :(

Reply:
Also, there should probably rarely be new native string literals being added.  Bytes or unicode?

Reply:
Got it. I think bs4 wants text on everything so I'll make these unicode literals.

I'll check the rest of the string literals but unfortunately I think some will have to be native strings since they're applied as keyword arguments.